### PR TITLE
feat(integration-platform):Prevent repeated publish requests

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective
When publishing a public integration, we want to create a new status to denote "Request Inprogress". And prevent further publish attempts while in progress. This should prevent repeated publish requests.

I also added more expansive error messages in the UI

## Migration
```
BEGIN;
--
-- Alter field status on sentryapp
--
COMMIT;
```
## UI
Screen Shot 2020-12-14 at 2 37 25 AM

### CODE REVIEW

File: src/sentry/static/sentry/app/components/modals/sentryAppPublishRequestModal.tsx [hunk 1]
```diff
@@ -143,8 +143,13 @@ export default class SentryAppPublishRequestModal extends React.Component<Props>
     this.props.closeModal();
   };
 
-  handleSubmitError = () => {
-    addErrorMessage(t('Request to publish %s fails.', this.props.app.slug));
+  handleSubmitError = err => {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/modals/sentryAppPublishRequestModal.tsx:146]: maybe you could add a type? :) 

`type OnSubmitError = Parameters<NonNullable<Form['props']['onSubmitError']>>;
`
```suggestion
  handleSubmitError = (error: OnSubmitError[0],  _instance: OnSubmitError[1], _id: OnSubmitError[2] ) => {
```

Comment (NisanthanNanthakumar) [reply src/sentry/static/sentry/app/components/modals/sentryAppPublishRequestModal.tsx:146]: @priscilawebdev But the error value has an `any` type on Form `onSubmitError` error arg. And all we want is just the error value. Do we need to type unused variables?

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/modals/sentryAppPublishRequestModal.tsx:146]: I understand, but I would still add the types. Maybe in the future, we can replace the type any of the forms with a better type and it will be automatically replicated here. But it's just a suggestion 😉  Frontend looks good.

File: src/sentry/models/sentryapp.py [hunk 2]
```diff
@@ -173,7 +177,6 @@ def save(self, *args, **kwargs):
     def is_installed_on(self, organization):
         return SentryAppInstallation.objects.filter(
             organization=organization, sentry_app=self,
-        ).exists()
```
Comment (markstory) [inline src/sentry/models/sentryapp.py:176]: Was this removal intentional?

Comment (NisanthanNanthakumar) [reply src/sentry/models/sentryapp.py:176]: @markstory arghh no! I merged the linter's suggestion from https://github.com/getsentry/sentry/pull/22644#discussion_r542307972
Good catch!

**Review by markstory — APPROVED** [review]
Migration looks good to me.

**Review by wedamija — COMMENTED** [review]
Migration is fine, just a no-op

Edit: lol, just saw mark already approved
